File: backend/app/Models/BatterySpec.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class BatterySpec extends Model
{
    public $timestamps = false;

    protected $fillable = [
        'product_id',
        'voltage',
        'ampere_hour',
        'cca',
        'battery_type',
        'terminal_position',
        'size_code',
    ];

    public function product(): BelongsTo
    {
        return $this->belongsTo(Product::class);
    }
}

File: flutter_app/lib/features/product/models/battery_spec.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'battery_spec.freezed.dart';
part 'battery_spec.g.dart';

/// Mirrors `frontend/types/product.type.ts` (`IBatterySpec`).
@freezed
abstract class BatterySpec with _$BatterySpec {
  const factory BatterySpec({
    required num voltage,
    required num ampere_hour,
    required num cca,
    required String battery_type,
    required String terminal_position,
    required String size_code,
  }) = _BatterySpec;

  factory BatterySpec.fromJson(Map<String, dynamic> json) =>
      _$BatterySpecFromJson(json);
}
